python/ray/data/_internal/datasource/bigquery_datasink.py

```diff
@@ -27,6 +27,7 @@ def __init__(
         dataset: str,
         max_retry_cnt: int = DEFAULT_MAX_RETRY_CNT,
         overwrite_table: Optional[bool] = True,
+        enable_list_inference: Optional[bool] = False,
```

Comment:
@francoisserra why don't we generalize this to allow passing in `LoadJobConfig` to provide flexibility in configuring how we write to BQ?

Reply:
Good call! 

I've missed the part that we're simply just uploading the Parquet right now. Alright, since its bound to Parquet let's not make API confusing and instead of whole `LoadJobConfig` just expose the `LoadJobConfig.ParquetOptions`